spec/unit/functions/epp_spec.rb

```diff
@@ -27,9 +27,14 @@
       expect(eval_template_with_args("<%= $phantom == dragos %>", 'phantom' => 'dragos')).to eq("true")
     end
 
-    it "can use values from the enclosing scope for defaults" do
+    it "can use values from the global scope for defaults" do
       scope['phantom'] = 'of the opera'
-      expect(eval_template("<%- |$phantom = $phantom| -%><%= $phantom %>")).to eq("of the opera")
+      expect(eval_template("<%- |$phantom = $::phantom| -%><%= $phantom %>")).to eq("of the opera")
+    end
+
+    it "can use values from the enclosing scope for defaults" do
```

Comment:
If this should work the same way as a function invocation, then it shouldn't use the enclosing scope at all. It should always use the global scope. But given the nature of templates (inlined in code) I guess this is one place where a function parameter declaration and a template parameter declaration may differ?